crates/editor/src/editor.rs

```diff
@@ -1230,16 +1228,9 @@ impl CompletionsMenu {
                                 None
                             };
 
-                        let color_swatch = if show_colors {
-                            Some(
-                                div()
-                                    .size_4()
-                                    .bg(completion.color().unwrap_or(gpui::transparent_black()))
-                                    .rounded(px(2.)),
-                            )
-                        } else {
-                            None
-                        };
+                        let color_swatch = completion
+                            .color()
+                            .map(|color| div().size_4().bg(color).rounded(px(2.)));
```

Comment:
Something that might be useful still is only showing the color column _if_ at least one of the completions is a color.

e.g. this could be just outside the loop
```rs
let show_colors = completions_guard.iter().any(|completion| {
    completion.lsp_completion.kind == Some(CompletionItemKind::COLOR)
});
```

Thoughts?

Reply:
I'm not sure we necessarily want that, as if you had a color that appeared off-screen then the rest of the completions would be seemingly indented for no reason.

And if you continued typing and filtered the color out there would be a visible jump of the rest of the completion items.

Reply:
Or more directly, do we want to reserve space so the text lines up because with this implementation it doesn't when some are colors and some are not:

<img width="244" alt="Screenshot 2024-10-03 at 14 34 35" src="https://github.com/user-attachments/assets/81d19580-284d-4b3d-86f6-1479e7a287a6">

Fine with either honestly but I think if they're lined up the completion list might be more scannable.

Reply:
> I'm not sure we necessarily want that, as if you had a color that appeared off-screen then the rest of the completions would be seemingly indented for no reason.
> 
> And if you continued typing and filtered the color out there would be a visible jump of the rest of the completion items.

Ah okay — 👍